src/main/java/com/es/phoneshop/model/features/recently/viewed/DefaultRecentlyViewedService.java

```diff
@@ -0,0 +1,51 @@
+package com.es.phoneshop.model.features.recently.viewed;
```

Comment:
Not the best package naming and structure, I'd recommend you to place it under com.es.phoneshop.model.product.feature package, the same for other classes of feature.